src/MSIdentityScaffolding/Microsoft.DotNet.MSIdentity/CodeReaderWriter/DocumentBuilder.cs

```diff
@@ -0,0 +1,533 @@
+                        {
+                            if (!string.IsNullOrEmpty(change.Block) && modifiedBlockSyntaxNode != null && parameterValues != null)
+                            {
+                                //CodeChange.Parent and CodeChange.Type go together.
+                                if (!string.IsNullOrEmpty(change.Parent) && !string.IsNullOrEmpty(change.Type))
+                                {
+                                    modifiedBlockSyntaxNode = AddCodeSnippetOnParent(change, modifiedBlockSyntaxNode, parameterValues);
+                                }
+
+                                //if there is no CodeChange.Parent, check if to InsertAfter a statement.
+                                else if (!string.IsNullOrEmpty(change.InsertAfter))
+                                {
+                                    modifiedBlockSyntaxNode = AddCodeSnippetAfterNode(change, modifiedBlockSyntaxNode, parameterValues);
+                                }
+                                //if there are no Parent or InsertAfter in CodeChange,
+                                //just insert statement at the end of the block
+                                else
+                                {
+                                    modifiedBlockSyntaxNode = AddCodeSnippetInPlace(change, modifiedBlockSyntaxNode, parameterValues);
+                                }
+                            }
+                        }
+                        //replace the BlockSyntax of a MethodDeclarationSyntax of a ClassDeclarationSyntax
+                        if (blockSyntaxNode != null && modifiedBlockSyntaxNode != null && modifiedClassDeclarationSyntax != null)
+                        {
+                            modifiedClassDeclarationSyntax = modifiedClassDeclarationSyntax.ReplaceNode(blockSyntaxNode, modifiedBlockSyntaxNode);
+                        }
+                    }
+                }
+            }
+            return modifiedClassDeclarationSyntax;
+        }
+
+        //add code snippet to parent node
+        internal BlockSyntax AddCodeSnippetOnParent(
+            CodeChange change,
+            BlockSyntax modifiedBlockSyntaxNode,
+            IDictionary<string, string> parameterValues)
+        {
+            string parentBlock = FormatCodeBlock(change.Parent, parameterValues).Trim(CodeSnippetTrimChars);
+            if (!string.IsNullOrEmpty(parentBlock))
+            {
+                //get the parent node to add CodeSnippet onto.
+                var parentNode = modifiedBlockSyntaxNode.DescendantNodes().Where(n => n is ExpressionStatementSyntax && n.ToString().Contains(parentBlock)).FirstOrDefault();
+                if (parentNode != null && parentNode is ExpressionStatementSyntax exprNode)
+                {
+                    //add a SimpleMemberAccessExpression to parent node.
+                    if ((change.Type?.Equals(CodeChangeType.MemberAccess)).GetValueOrDefault(false))
```

Comment:
>CodeChangeType.MemberAccess [](http://example.com/codeflow?start=45&length=27)

could you apply Equals on this one to avoid null on change.Type?

Reply:
fixed and moved to https://github.com/dotnet/Scaffolding/pull/1593/